modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -521,31 +520,33 @@ public static void handleFailure(HTTPCarbonMessage requestMessage, BallerinaConn
         String errorMsg = ex.getMessage();
         log.error(errorMsg);
         ErrorHandlerUtils.printError(ex);
-        if (statusCode == 404) {
-            sendOutboundResponse(requestMessage, createErrorMessage(errorMsg, statusCode));
-        } else {
-            sendOutboundResponse(requestMessage, createErrorMessage("", statusCode));
-        }
+        sendOutboundResponse(requestMessage, createErrorMessage(errorMsg, statusCode));
     }
 
     public static HTTPCarbonMessage createErrorMessage(String payload, int statusCode) {
         HTTPCarbonMessage response = HttpUtil.createHttpCarbonMessage(false);
         response.waitAndReleaseAllEntities();
-        response.addHttpContent(new DefaultLastHttpContent(Unpooled.wrappedBuffer(payload.getBytes())));
-        setHttpStatusCodes(payload, statusCode, response);
+        if (payload != null) {
+            payload = lowerCaseTheFirstLetter(payload);
+            response.addHttpContent(new DefaultLastHttpContent(Unpooled.wrappedBuffer(payload.getBytes())));
+        }
+        setHttpStatusCodes(statusCode, response);
 
         return response;
     }
 
-    private static void setHttpStatusCodes(String payload, int statusCode, HTTPCarbonMessage response) {
+    private static String lowerCaseTheFirstLetter(String payload) {
+        char[] characters = payload.toCharArray();
+        characters[0] = Character.toLowerCase(characters[0]);
```

Comment:
Are we sure length of payload is not 0 here ?

Reply:
Well, this payload is taken from BallerinaConnectorException and when you create BallerinaConnectorException we must give an informative error message as it is a common exception. But it was not enforced by the its constructors. hence, I update that class. Therefore, theoretically there shouldn't be an empty string.  

Anyways, just to be on the safe-side. I have checked for an empty string as well.